src/domain/usecases/ImportUsersUseCase.ts

```diff
@@ -31,21 +31,24 @@ export class ImportUsersUseCase implements UseCase {
     constructor(private userRepository: UserRepository) {}
 
     public execute({ users }: ImportUsersUseCaseOptions): FutureData<void> {
-        const usernameList = users.map(user => user.username);
-        return Future.join2(
-            this.userRepository.listAll({ filters: { "userCredentials.username": ["in", usernameList] } }),
-            this.userRepository.getCurrent()
-        ).flatMap(([usersFromDB, currentUser]: [User[], User]) => {
-            const hasRequiredFields = UserLogic.validateHasRequiredFields(users);
-            if (!hasRequiredFields)
-                return Future.error("All users must have at least one Organisation Unit, Role and Group");
+        return Future.futureMap(_.chunk(users, chunkSize), userChunk => {
```

Comment:
Nothing wrong with `futureMap` but I think in this context we should do sequential requests [instead parallel requests](https://github.com/EyeSeeTea/user-extended-app/blob/development/src/domain/entities/Future.ts#L101). This way it's easier to debug errors and we are sure next requests are not executed.

Something like this:

```ts
 return Future.sequential(_.chunk(users, chunkSize).map((usersChunk) => {
   ... 
 })
).toVoid();
```